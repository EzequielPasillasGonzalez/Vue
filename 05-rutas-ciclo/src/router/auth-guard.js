const isAuthenticatedGuard = (to, from, next) => {
    //console.log({to, from, next});

    return new Promise( (reolve, reject) => {
        const random = Math.random() * 100
        
        if(random > 50){
            console.log('Esta autenticado');
            next()
        }else{
            console.log('Es bloqueado por isAuthenticatedGuard', random);
            next({name: 'pokemon-home'})
        }
    })

    
}

export default isAuthenticatedGuard;