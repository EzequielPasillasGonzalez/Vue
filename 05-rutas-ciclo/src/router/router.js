import {createRouter, createWebHashHistory}  from 'vue-router'
import isAuthenticatedGuard from './auth-guard'

const routes = [
    {
        path     : '/',
        redirect : '/pokemon',
    },    
    {
        path: '/pokemon',
        name: 'pokemonLayout',
        component : () => import( /* webpackChunkName: "PokemonLayout" */ '@/modules/pokemon/layouts/PokemonLayout.vue'),
        children: [
        {   
            path      : 'home', 
            name      : 'pokemon-home',
            component : () => import( /* webpackChunkName: "ListPage" */ '@/modules/pokemon/pages/ListPagePokemon.vue')
        }, //? Cargar de manera precezosa },
        {   
            path      : 'about',
            name      : 'pokemon-about',
            component : () => import( /* webpackChunkName: "AboutPage" */ '@/modules/pokemon/pages/AboutPagePokemon.vue')
        }, //? Cargar de manera precezosa
        {   
            path      : 'pokemonid/:id',
            name      : `pokemon-id`, //? Se recomienda navegar por el name en las rutas
            component : () => import( /* webpackChunkName: "PokemonPage" */ '@/modules/pokemon/pages/PokemonPagePokemon.vue'), //? Cargar de manera precezosa  },
            props     : (route) => {
                const id = Number(route.params.id)
                return isNaN(id) ? {id : 1} : { id } //? Pasarlo como una propertie del componente y poder usarlo alli                        
            }
        },
        {
            path: '', //? Si esta vacio el path, es el primer componente que se muestra al llegar a la ruta
            redirect: { name: 'pokemon-about' },
        }
        ]
    },
    {
        
        path: '/dbz',
        name: 'dbzLayout',
        beforeEnter: [ isAuthenticatedGuard ], // Se aplica un Gap solo para estas rutas
        component : () => import( /* webpackChunkName: "DBZ Layout" */ '@/modules/dbz/layouts/DragonBallLayout.vue'),
        children: [
            {
                //**Layout */
                path: 'dbz',
                name: 'dbz-characters',
                component : () => import( /* webpackChunkName: "DBZ Characters" */ '@/modules/dbz/pages/Characters.vue'),
            },
            {
                //**About */
                path: 'about',
                name: 'dbz-about',
                component : () => import( /* webpackChunkName: "DBZ About" */ '@/modules/dbz/pages/About.vue'),
            },
            {
                path: '', //? Si esta vacio el path, es el primer componente que se muestra al llegar a la ruta
                redirect: { name: 'dbz-characters' },
            }
        ]
    },    
    {   
        path      : '/:pathMatch(.*)*',
        name      : 'NotFound',
        component : () => import( /* webpackChunkName: "No Page Found" */ '@/modules/shared/pages/NoPageFound.vue')  //? Cargar de manera precezosa },        
    },
]

const router = createRouter({
    // 4. Provide the history implementation to use. We are using the hash history for simplicity here.
    history: createWebHashHistory(),
    routes, // short for `routes: routes`
})


// Guard Global - Sincrono
// router.beforeEach((to, from, next) => {
//     console.log({to, from, next});

//     const random = Math.random() * 100

//     if( random > 50){
//         console.log(`Autenticado`);
//         next()
//     }else{
//         console.log(random, 'Bloqueado por el beforeEach Guard');
//         next({name: 'pokemon-home'})
//     }

//     //next()
// })


//! Importante aplicar
// const canAccess = () => { // Protecciones para que no puedan pasar - Login probable uso
//     return new Promise( (resolve, reject) => {
//         const random = Math.random() * 100

//         if( random > 50){
//             console.log(`Autenticado - canAccess`);
//             resolve(true)
//         }else{
//             console.log(random, 'Bloqueado por el beforeEach Guard - canAccess');
//             reject()
//         }
//     })
// }

// router.beforeEach( async (to,from,next) => {
    

//     try {
//         const authorized = await canAccess()
//         next()
//     } catch (error) {
//         next({name: 'pokemon-home'})
//     }
// })

export default router;