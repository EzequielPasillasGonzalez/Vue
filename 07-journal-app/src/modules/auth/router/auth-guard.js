import store from '@/store'

const isAuthenticated = async (to, from, next) => {

    console.log(to, from, next);

    const { ok } = await store.dispatch('auth/checkAuthtentication')

    if(ok){
        next()
    }else{
        next({name: 'login'})
    }

    
}

export default isAuthenticated