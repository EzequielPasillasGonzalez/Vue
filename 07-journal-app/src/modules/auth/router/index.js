export default {
    name: 'auth',
    component: () =>  import(/* webpackChunckName: "authLayout" */ '@/modules/auth/layouts/AuthLayout.vue'),
    children: [
        {
            path: '',
            name: 'login',
            component: () => import(/* webpackChunckName: "login" */ '@/modules/auth/views/LoginView.vue'),
        },
        {
            path: '/register',
            name: 'register',
            component: () => import(/* webpackChunckName: "register" */ '@/modules/auth/views/RegisterView.vue'),
        }
    ]
}