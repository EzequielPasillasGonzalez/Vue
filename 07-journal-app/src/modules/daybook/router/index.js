export default {
    name: 'dayBook',    
    component: () => import(/* webpackChunckName: "daybook" */ '@/modules/daybook/layouts/DayBookLayout.vue'),
    children: [
        {
            path: '',
            name: 'no-entry',    
            component: () => import(/* webpackChunckName: "daybook-no-entry" */ '@/modules/daybook/views/NoEntrySelectedView.vue'),
        },
        {
            path: ':id',
            name: 'entry',    
            component: () => import(/* webpackChunckName: "daybook-entry" */ '@/modules/daybook/views/EntryView.vue'),
        },
    ]
}