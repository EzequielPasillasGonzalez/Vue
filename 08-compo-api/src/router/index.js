import { createRouter, createWebHashHistory } from 'vue-router'


const routes = [
  {
    path: '/',
    name: 'home',
    component: () => import(/* webpackChunkName: "home" */ '../views/HomeView.vue')
  },
  {
    path: '/about',
    name: 'about',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () => import(/* webpackChunkName: "about" */ '../views/AboutView.vue')
  },
  {
    path: '/counter',
    name: 'counter',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () => import(/* webpackChunkName: "counter" */ '../views/CounterView.vue')
  },
  {
    path: '/users',
    name: 'users',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () => import(/* webpackChunkName: "user" */ '../views/UserView.vue')
  },
  {
    path: '/pokemon-search',
    name: 'pokemon-search',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () => import(/* webpackChunkName: "pokemon-search" */ '../views/SearchPokemonView.vue')
  },
  {
    path: '/pokemon/:id',
    name: 'pokemon-id',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () => import(/* webpackChunkName: "pokemon-id" */ '../views/PokemonView.vue')
  },
  {
    path: '/todo',
    name: 'todo',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () => import(/* webpackChunkName: "todoVuex" */ '../views/TodoVuexView.vue')
  },
  {
    path: '/slots',
    name: 'slots',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () => import(/* webpackChunkName: "Slots" */ '../views/CustomSlotsView.vue')
  }
]

const router = createRouter({
  history: createWebHashHistory(),
  routes
})

export default router
