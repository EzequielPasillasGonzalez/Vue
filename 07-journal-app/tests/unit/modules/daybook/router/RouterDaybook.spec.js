import { shallowMount } from "@vue/test-utils"

import router from "@/modules/daybook/router"

describe('Pruebas en Router del Daybook', () => {


    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(router)
    })

    test('El router debe de tener esta configuracion', async () => {

        expect(router).toMatchObject({
            name: 'dayBook',
            component: expect.any( Function ),
            children: [
                {
                    path: '',
                    name: 'no-entry',
                    component: expect.any( Function ),
                },
                {
                    path: ':id',
                    name: 'entry',
                    component: expect.any( Function ),
                    props: expect.any( Function )                    
                },
            ]
        })

        // expect(  (await router.children[0].component()).default.name ).toBe('NoEntrySelected')
        // expect(  (await router.children[1].component()).default.name ).toBe('EntryView')

        const promiseRoutes = []

        router.children.forEach( child => promiseRoutes.push(child.component()) )

        const routes = (await Promise.all( promiseRoutes )).map( r => r.default.name)
        


    })


    test('Debe de retornar el Id de la ruta', () => {

        const route = {
            params: {
                id: 'ABC-132'
            }
        }

        // expect(router.children[1].props(route)).toEqual({id: 'ABC-132'});

        const entryRoute = router.children.find( route =>  route.name === 'entry')

        expect(entryRoute.props(route)).toEqual({id: 'ABC-132'});


    })

})