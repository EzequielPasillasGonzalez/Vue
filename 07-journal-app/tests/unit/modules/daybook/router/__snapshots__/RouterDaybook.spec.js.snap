// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Router del Daybook El router debe de tener esta configuracion 1`] = `""`;
